snapcraft/plugins/python.py

```diff
@@ -238,8 +242,11 @@ def _get_constraints(self):
             if isurl(self.options.constraints):
                 constraints = {self.options.constraints}
             else:
-                constraints = {os.path.join(
-                    self.sourcedir, self.options.constraints)}
+                constraints = {self._find_file(
+                    filename=self.options.constraints)}
+                if not constraints:
+                    # TODO snapcraft owned message
+                    raise FileNotFoundError(self.options.constraints)
```

Comment:
This is new behavior, is it not? Why not define the error class here? Also would be nice to see tests covering these cases.

Reply:
we don't have any place to add them for plugins, this is urgent and needs to come out today (that is why we have a TODO)

Reply:
That [hasn't stopped us in the past](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/catkin.py#L103). Your call, of course. No tests either, then?